docs/docs/production-deployment.md

```diff
@@ -0,0 +1,161 @@
+
+Note that deployments on Kubernetes can utilize The [Pomerium Ingress Controller](/docs/k8s/ingress.md) to simplify configuration.
+
+### Authenticate
+
+The suggested practice is to use the Pomerium Proxy service to load-balance Authenticate. Alternately, you could use an independent Layer 4 or Layer 7 load balancer, but this increases complexity.
+
+
+### Authorize and Databroker
+
+You do **not** need to provide a load balancer in front of Authorize and Databroker services. Both utilize GRPC, and thus has special requirements if you should choose to use an external load balancer. GRPC can perform client based load balancing, and in most configurations is the best architecture.
+
+By default, Pomerium gRPC clients will automatically connect to all IPs returned by a DNS query for the name of an upstream service. They will then regularly re-query DNS for changes to the Authorize or Databroker service cluster. Health checks and failover are automatic.
+
+You can also define multiple domain names for Databroker or Authorize services with the [`databroker_service_url`](/reference/readme.md#data-broker-service-url) and [`authorize_service_urls`](/reference/readme.md#authorize-service-url) shared config keys.
+
+## High Availability
+
+As mentioned in [scaling](#scaling), Pomerium components themselves are stateless and support horizontal scale out for both availability and performance reasons.
+
+A given service type does not require communication with its peer instances to provide high availability. Eg, a Proxy service instance does not communicate with Proxy instances.
+
+Regardless of the service mode, it is recommended you run at least 2 instances of Pomerium with as much physical and logical separation as possible. For example, in Cloud environments, you should deploy instances of each service to at least 2 different zones. On-prem environments should deploy >=2 instances to independent hardware.
+
+Ensure that you have enough spare capacity to handle the scope of your failure domains.
+
+::: warning
+Multiple replicas of Databroker or all-in-one service are only supported with [external storage](/docs/topics/data-storage.md) configured
+:::
+
+## SSL/TLS Certificates
+
+Pomerium utilizes TLS end to end, so the placement, certificate authorities and covered subjects are critical to align correctly.
+
+In a typical deployment, a minimum of two certs are required:
+
+- A wildcard certificate which covers the external `from` names, the Proxy service's external name and the Authenticate service's external name
+ - Must be trusted by user browsers or clients
+ - Must cover subject names from the user perspective
+- A certificate which covers the Authorize service's name
+ - Must be trusted by the Proxy
+ - Must cover the subject name from the Proxy's perspective
+
```

Comment:
No actual change necessary, but is there a reason the above list is in that order? It goes
- A
- Must
- Must
- A 
- Must
- Must

I'm suggesting put the A's together.

Reply:
The Musts should be nested. [Adjusted](https://github.com/pomerium/pomerium/pull/3129/files#r835496299).